lib/system/gcmalloc.nim

```diff
@@ -0,0 +1,540 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2012 Andreas Rumpf
+#        (c) Copyright 2021 Jacek Sieka
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+# Malloc-based allocator for the GC - similar to what's found in alloc.nim
+# without all the fancy small chunk allocation stuff - enable with -d:useMalloc
```

Comment:
maybe rename `useMalloc`, so ARC or ORC can benefit from this allocator.

Reply:
you mean the module? I'm not against though I don't have enough insight into orc/arc really - I imagine it could be simplified a bit for those use cases though if they are no longer stack-scanning, no?